src/Groups/matrices/form_group.jl

```diff
@@ -0,0 +1,629 @@
+            A[i,j] = em(a)
+            B[i,j] = em(b)
+         end
+         MM = zero_matrix(F0, n^2, n^2)
+         pos_r = 1
+         pos_c = 1
+
+         # in the vector of solutions, the coefficients are ordered as:
+         # first the x_ii, then the x_ij (i<j) and finally the y_ij (i<j)
+
+         # coefficients for the x_ii
+         for i in 1:n
+            for h in 1:n
+               MM[pos_r,pos_c] = A[i,h]^2+A[i,h]*B[i,h]+Nw*B[i,h]^2
+               pos_c +=1
+            end
+            for h in 1:n, k in h+1:n
+               MM[pos_r,pos_c] = A[i,k]*B[i,h]+A[i,h]*B[i,k]
+               pos_c +=1
+            end
+            for h in 1:n, k in h+1:n
+               MM[pos_r,pos_c] = A[i,h]*A[i,k]+A[i,k]*B[i,h]+Nw*B[i,k]*B[i,h]
+               pos_c +=1
+            end
+            pos_r +=1
+            pos_c =1
+         end
+
+         # coefficients for the x_ij
+         for i in 1:n, j in i+1:n
+            for h in 1:n
+               MM[pos_r,pos_c] = A[i,h]*A[j,h]+A[i,h]*B[j,h]+Nw*B[i,h]*B[j,h]
+               pos_c +=1
+            end
+            for h in 1:n, k in h+1:n
+               MM[pos_r,pos_c] = A[i,h]*A[j,k]+A[i,h]*B[j,k]+A[i,k]*B[j,h]+A[i,k]*A[j,h]+Nw*(B[i,h]*B[j,k]+B[i,k]*B[j,h])
+               pos_c +=1
+            end
+            for h in 1:n, k in h+1:n
+               MM[pos_r,pos_c] = Nw*(A[i,h]*B[j,k]+A[j,k]*B[i,h]+B[i,k]*B[j,h]+A[i,k]*B[j,h]+A[j,h]*B[i,k])+A[i,k]*A[j,h]+A[i,k]*B[j,h]
+               pos_c +=1
+            end
+            pos_r +=1
+            pos_c =1
+         end
+
+         # coefficients for the y_ij
+         for i in 1:n, j in i+1:n
+            for h in 1:n
+               MM[pos_r,pos_c] = A[j,h]*B[i,h]+A[i,h]*B[j,h]
+               pos_c +=1
+            end
+            for h in 1:n, k in h+1:n
+               MM[pos_r,pos_c] = A[j,k]*B[i,h]+A[i,h]*B[j,k]+A[j,h]*B[i,k]+A[i,k]*B[j,h]
+               pos_c +=1
+            end
+            for h in 1:n, k in h+1:n
+               MM[pos_r,pos_c] = A[i,h]*A[j,k]+A[i,k]*A[j,h]+A[j,k]*B[i,h]+A[i,k]*B[j,h]+Nw*(B[i,h]*B[j,k]+B[i,k]*B[j,h])
+               pos_c +=1
+            end
+            pos_r +=1
+            pos_c =1
+         end
+ 
+         for i in 1:n^2
+            MM[i,i] -=1
+         end
+         push!(M,MM)
+      end
+   end
+
+   n_gens,K = nullspace(block_matrix(length(M),1,M))
+   N = div(n*(n-1),2)
+   L = T[]
+
+   for r in 1:n_gens
+      sol = zero_matrix(F,n,n)
+      for i in 1:n
+         sol[i,i] = e(K[i,r])
+      end
+      pos = n+1
+      if isodd(q)
+         for i in 1:n, j in i+1:n
+            sol[i,j] = e(K[pos,r])+w*e(K[pos+N,r])       # sol[i,j] = a_ij + w*b_ij
+            sol[j,i] = e(K[pos,r])-w*e(K[pos+N,r])
+            pos +=1
+         end
+      else
+         for i in 1:n, j in i+1:n
+            sol[i,j] = e(K[pos,r])+w*e(K[pos+N,r])       # sol[i,j] = a_ij + w*b_ij
+            sol[j,i] = e(K[pos,r])+e(K[pos+N,r])+w*e(K[pos+N,r])
```

Comment:
So... the same as this?
```suggestion
            sol[j,i] = e(K[pos,r])+(1+w)*e(K[pos+N,r])
```

I am lost, why is this code right?

Reply:
The matrix `sol` must be hermitian, so `sol[j,i]` is equal to `sol[i,j]^q` for every `i,j`. Now, `(a_ij+w*b_ij)^q = a_ij+w^q*b_ij`. We built `w` in order to have `w^q=w+1`. It follows that if `sol[i,j]=a_ij+w*b_ij`, then `sol[j,i] = a_ij+b_ij+w*b_ij`.

Reply:
These things (and much more) should be explained in some code comments